docs/Homebrew-on-Linux.md

```diff
@@ -47,8 +47,7 @@ If you're using an older distribution of Linux, installing your first package wi
 
 ## Requirements
 
-- **GCC** 4.7.0 or newer
```

Comment:
If we're dropping this then `brew postinstall gcc` should probably fail a bit more gracefully, and perhaps display a warning that the C/C++ compilers will not be able to build executables.

Reply:
Our usage of the host GCC in `brew postinstall gcc` might be avoidable if there is a reliable way to find the directory containing `crti.o` and related files without relying on the host compiler toolchain.  I have not looked carefully into how to do this but it's the only place in all of homebrew-core where the host compiler is forcibly invoked and it would be great to get rid of it if possible.

Reply:
My understanding is that you don't even have a `crti.o` unless you install GCC (or some similar package) using your system package manager.

My suggestion was that we shouldn't bother looking for it if there is no host GCC. Using `gcc-*` or `g++-*` will not work, but you'll be able to use the runtime libraries.

Reply:
I think the `crti.o` files are actually distributed as part `glibc` though, not GCC - at least I can see that is the case on Arch Linux based on the file list for `glibc`.  But nonetheless in the interim I agree that we should check if `/usr/bin/cc` exists and give some informative message indicating that the run time libraries will work but the compilers will not.

Reply:
> My understanding is that you don't even have a `crti.o` unless you install GCC (or some similar package) using your system package manager.

crti.o is a glibc development file (`libc6-dev`), which makes sense that we require glibc headers and libraries when we are compiling against system glibc.

The only role GCC has is to search for this file, so the version of GCC itself is largely irrelevant as long as it's configured against the system glibc directories which we assume the one in `/usr/bin` to be. The usage of GCC does sounds a little excessive, but I'm not sure what an alternative central sources of truth would be in terms of finding where `crti.o` is located. If there is one: great.

If we ultimately will always need GCC for this search procedure, we can simplify the requirement to just "GCC" perhaps. Though in the case that our glibc is used, a system GCC is not required AFAIK.